CRAYSAT-1620: Changes for setting up SAT on an external system (#59)
Cray-HPE/docs-sat: docs/install.md

   SAT CSM Python dependency packages are not currently distributed publicly as
   source packages or binary distributions, so they must be installed from
   their source code hosted on GitHub. Furthermore, to install the `cray-product-catalog`
   Python package, you must first clone it locally. Execute the following six steps to
   modify the SAT CSM Python dependencies so they can be installed from their source code.

   1. Clone the source code for `cray-product-catalog`.

      ```screen
      (venv) $ git clone --branch v1.6.0 https://github.com/Cray-HPE/cray-product-catalog
      ```

   1. Create a file named `.version` in the `cray-product-catalog` directory
      containing the version of `cray-product-catalog`.

      ```screen
      (venv) $ echo 1.6.0 > cray-product-catalog/.version
      ```

   1. Open the "locked" requirements file in a text editor.

      ```screen
      (venv) $ vim sat/requirements.lock.txt
      ```

   1. Change the line containing `cray-product-catalog` to refer to the local path
      to `cray-product-catalog`. It should read as follows:

      ```screen
      ./cray-product-catalog
      ```

>    1. For versions of SAT greater than 3.19, change the line containing `csm-api-client` to

Reviewer: ```suggestion
   1. For versions of SAT newer than 3.19, change the line containing `csm-api-client` to
```

   modify the SAT CSM Python dependencies so they can be installed from their source code.

   1. Clone the source code for `cray-product-catalog`.

      ```screen
      (venv) $ git clone --branch v1.6.0 https://github.com/Cray-HPE/cray-product-catalog
      ```

   1. In the `cray-product-catalog` directory, create a file named `.version`
      that contains the version of `cray-product-catalog`.

      ```screen
      (venv) $ echo 1.6.0 > cray-product-catalog/.version
      ```

   1. Open the "locked" requirements file in a text editor.

      ```screen
      (venv) $ vim sat/requirements.lock.txt
      ```

   1. Update the line containing `cray-product-catalog` so that it reflects the local path
      to `cray-product-catalog`.

      It should read as follows:

      ```screen
      ./cray-product-catalog
      ```

   1. For versions of SAT newer than 3.19, change the line containing `csm-api-client` to
      read as follows:

      ```screen
      csm-api-client@git+https://github.com/Cray-HPE/python-csm-api-client@release/1.1
      ```

   1. (Optional) Confirm that `requirements.lock.txt` is modified as expected.

>       **Note:** Only if installing a version newer than 3.19 will you see `csm-api-client`.

Reviewer: What about this wording? I think it keeps your original intent.
```suggestion
      **Note:** You will only see `csm-api-client` if you are installing a version newer than 3.19.
```

Author: Sounds good, I'll take it!

>       **Note:** If installing a version newer than 3.19, you should only see `csm-api-client`.

Reviewer 2: I think this note is wrong. I think if installing 3.19, you will only see `cray-product-catalog`. If installing newer than 3.19, you will see both.

Perhaps the "only" is just in the wrong place. It would make sense if the sentence stated "Only if installing a version newer than 3.19...."

Reviewer 2: Oh yeah, I see that it was correct in the first version and then was misinterpreted and changed in an earlier review. Perhaps my rewording in the first paragraph is more clear.

Author: > Perhaps the "only" is just in the wrong place. It would make sense if the sentence stated "Only if installing a version newer than 3.19...."

Oops, yes that is correct. It originally said "You should only see `csm-api-client` if installing a version greater than 3.19," which was ambiguous.

Author: I pushed a commit addressing this comment. I still think it's a little awkward sounding if you or @jld-dehn-hpe have further ideas for rewording it. I wanted to make it really unambiguous that "only" is referring to "only if you are installing >3.19" not "only `csm-api-client`".

Reviewer: @ekamin-hpe, I suggested a different way of wording this. You'll have to see if you like it better.

>       **Note:** You will only see `csm-api-client` if you are installing a version newer than 3.19.

Reviewer: ```suggestion
      **Note:** If installing version 3.19, you will only see `csm-api-client`.
```

Author: This would be incorrect - (a) should say "versions newer than 3.19" (b) this makes it sound like you only see `csm-api-client` when I'm actually trying to say "versions newer than 3.19 implies you see `csm-api-client`" if that makes sense. This is pretty much what you originally suggested in: https://github.com/Cray-HPE/docs-sat/pull/59#discussion_r1023339178

Just to be more clear about what I'm trying to say:

Versions newer than 3.19: You will see both `csm-api-client` and `cray-product-catalog`.
Versions 3.19 and older: You will only see `cray-product-catalog`, you will not see `csm-api-client`.

Reviewer: Thanks for the explanation! I was having some trouble connecting the dots. I think something like the following might be more clear, with the emphasis of "only" on the versions that have just the one option.

```suggestion
      **Note:** For versions newer than 3.19, you will see both `cray-product-catalog` and `csm-api-client`. For version 3.19 and older, you will only see 'cray-product-catalog`.
```

Author: Nice, yup that is good with me, thanks for the suggestion.
